src/modules/acbs/acbs-facility-fixed-fee.service.test.ts

```diff
@@ -94,4 +120,142 @@ describe('AcbsFacilityFixedFeeService', () => {
+      await expect(createFixedFeeForFacilityPromise).rejects.toThrow(`Facility with identifier ${facilityIdentifier} was not found by ACBS.`);
+      await expect(createFixedFeeForFacilityPromise).rejects.toHaveProperty('innerError', axiosError);
+    });
+
+    it('throws an AcbsBadRequestException if ACBS responds with a 400 that is a string that does not contain "The facility not found"', async () => {
+      const axiosError = new AxiosError();
+      const errorString = valueGenerator.string();
+      axiosError.response = {
+        data: errorString,
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      when(httpServicePost)
+        .calledWith(...expectedHttpServicePostArgs)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const createFixedFeeForFacilityPromise = service.createFixedFeeForFacility(
+        portfolioIdentifier,
+        facilityIdentifier,
+        acbsRequestBodyToCreateFacilityFixedFee,
+        idToken,
+      );
+
+      await expect(createFixedFeeForFacilityPromise).rejects.toBeInstanceOf(AcbsBadRequestException);
+      await expect(createFixedFeeForFacilityPromise).rejects.toThrow(`Failed to create a fixed fee for facility ${facilityIdentifier} in ACBS.`);
+      await expect(createFixedFeeForFacilityPromise).rejects.toHaveProperty('innerError', axiosError);
+      await expect(createFixedFeeForFacilityPromise).rejects.toHaveProperty('errorBody', errorString);
+    });
+
+    it('throws an AcbsBadRequestException if ACBS responds with a 400 that is not a string', async () => {
+      const axiosError = new AxiosError();
+      const errorBody = { errorMessage: valueGenerator.string() };
+      axiosError.response = {
+        data: errorBody,
+        status: 400,
+        statusText: 'Bad Request',
+        headers: undefined,
+        config: undefined,
+      };
+
+      when(httpServicePost)
+        .calledWith(...expectedHttpServicePostArgs)
+        .mockReturnValueOnce(throwError(() => axiosError));
+
+      const createFixedFeeForFacilityPromise = service.createFixedFeeForFacility(
+        portfolioIdentifier,
+        facilityIdentifier,
+        acbsRequestBodyToCreateFacilityFixedFee,
+        idToken,
+      );
+
+      await expect(createFixedFeeForFacilityPromise).rejects.toBeInstanceOf(AcbsBadRequestException);
+      await expect(createFixedFeeForFacilityPromise).rejects.toThrow(`Failed to create a fixed fee for facility ${facilityIdentifier} in ACBS.`);
+      await expect(createFixedFeeForFacilityPromise).rejects.toHaveProperty('innerError', axiosError);
+      await expect(createFixedFeeForFacilityPromise).rejects.toHaveProperty('errorBody', JSON.stringify(errorBody));
+    });
+
+    it('throws an AcbsUnexpectedException if ACBS responds with an error code that is not 400', async () => {
```

Comment:
Can we add an assertion where the fixed fee record already exists please.

Reply:
Good idea.
I added ACBS service test for ACBS error "FixedFee exists" and same test in api-test. Also created "know error" object for 400 "FixedFee exists".